src/esbmc/bmc.cpp

```diff
@@ -770,95 +786,82 @@ smt_convt::resultt bmct::multi_property_check(
-          local_eq, runtime_solver, goto_trace, is_compact_trace);
-
-        // Store the comment and location of the assertion
-        // to avoid double verifying the claims that are already verified
-        std::string cmt_loc = "";
-
-        for(const auto &step : goto_trace.steps)
-          if(step.type == goto_trace_stept::ASSERT)
-          {
-            // since we only handle one claim at a time
-            // we will/should not overwrite the loc
-            assert(cmt_loc == "");
-            std::string loc;
-            if(step.pc->location.is_nil())
-              loc = "nil";
-            else
-              loc = step.pc->location.as_string();
-
-            // we use the "comment + location" to distinguish each claim
-            // - locaiton: ideally, the loc is enough for distinguishuing claims.
-            // - comment: we add a claim number prefix in comment. Note that the unwinding assertions which are not processed by goto-cov.
-            // Another reason is that we need to display this info in goto-coverage-claims
-
-            cmt_loc = step.comment + "\t" + loc;
-          }
-
-        bool is_unverified = false;
-
-        if(is_goto_cov)
-        {
-          reached_mul_claims.emplace(cmt_loc);
-          is_unverified = true;
-        }
-        else
+      goto_tracet goto_trace;
+      build_goto_trace(local_eq, runtime_solver, goto_trace, is_compact_trace);
+
+      // Store the comment and location of the assertion
+      // to avoid double verifying the claims that are already verified
+      std::string cmt_loc = "";
+
+      for(const auto &step : goto_trace.steps)
+        if(step.type == goto_trace_stept::ASSERT)
         {
-          // the ins is true if the element was actually inserted
-          auto [it, ins] = reached_claims.emplace(cmt_loc);
-          is_unverified = ins;
+          // since we only handle one claim at a time
+          // we will/should not overwrite the loc
+          assert(cmt_loc == "");
+          std::string loc;
+          if(step.pc->location.is_nil())
+            loc = "nil";
+          else
+            loc = step.pc->location.as_string();
+
+          // we use the "comment + location" to distinguish each claim
+          // e.g. "Claim x: ... location line y"
+          // x is unique. However, the unwinding asserts do not have these Claim x prefixes.
+          // Therefore we add the location behind, as the line number y for each unwinding assert is different.
+          cmt_loc = step.comment + "\t" + loc;
         }
 
-        if(is_unverified || options.get_bool_option("keep-verified-claims"))
+      bool is_unverified = false;
+
+      if(is_goto_cov)
+      {
+        reached_mul_claims.emplace(cmt_loc);
+        is_unverified = true;
+      }
+      else
+      {
+        // the ins is true if the element was actually inserted
+        auto [it, ins] = reached_claims.emplace(cmt_loc);
+        is_unverified = ins;
+      }
```

Comment:
I suppose this `if`/`else` could be simplified to only call `.emplace` once.

Reply:
Oh, indeed. I didn't realize those were two different ones. Thanks, that's fine then, no change is necessary.